framework/src/st_things/things_stack/inc/things_api.h

```diff
@@ -0,0 +1,117 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef _THINGS_API_H_
+#define _THINGS_API_H_
+
+#include <stdint.h>
+#include "things_types.h"
+#include "things_resource.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif							// __cplusplus
+
+#define TYPE_INT_64 "int64"
+#define TYPE_BOOL   "boolean"
+#define TYPE_DOUBLE "double"
+#define TYPE_STRING "string"
+#define TYPE_BYTE    "byte"
+
+#define MAX_IPADDR           (16)
+#define OIC_UUID_STRING_SIZE (37)
+
+const char *things_get_sw_version();
+const char *things_get_device_id_string();
+int things_set_device_name(int dev_num, char *name);
+
+int things_initialize_stack(const char *json_path, bool *easysetup_completed);
+int things_deinitialize_stack();
+int things_start_stack();
+
+typedef void (*things_reset_result_func_type)(int result);
+
+typedef int (*things_reset_confirm_func_type)(things_reset_result_func_type *func_carrier, things_es_enrollee_reset_e reset_type);
+
+int things_register_confirm_reset_start_func(things_reset_confirm_func_type func);
+
+int things_return_user_opinion_for_reset(int b_reset_start);
+
+int things_reset(void *remote_owner, things_es_enrollee_reset_e reset_type);
+
+int things_stop(void);
```

Comment:
This stop API is not used in Tizen RT because things stack has to be available 24/7.
Why don't you remove this API?

Reply:
modified